### ISSUE TITLE
Create isobands inside entire polygon

### ISSUE THREAD

**Issue by mumispb** [Turfjs/turf#2444]
I have a polygon and some measurements made inside that polygon and I would like to create isobands for the entirety of it. However, it seems interpolate only runs inside the bounding box that it creates automatically and it only encapsulates the points that I have provided with values. How could I expand it to the entire polygon?
![image](https://github.com/Turfjs/turf/assets/22329056/6763214e-2f47-4d38-9cd7-11f6708517af)

This is the exact same issue in https://github.com/Turfjs/turf/issues/2297. On that issue, the author has provided a working fiddle for his case https://jsfiddle.net/doug_b/xbd6Lf74/3/.

Turf.js version: 6.5.0

**Comment by lemonig** [comment]
oh! dear.
 I also encountered the same problem, a year has passed, have you solved it?
I read the source code of truf.js, `interpolate `method is bounded by known points, I think it should be filled with polygon as the boundary to achieve interpolation, if you have any methods or changes to solve this problem, please let me know, thank you!
![image](https://github.com/user-attachments/assets/1ba6bbdb-b4c1-4c00-9c36-2aaac777cd94)

![image](https://github.com/user-attachments/assets/22174460-61f2-4508-a722-f6d46901af3d)

**Comment by lemonig** [comment]
> HI @mumispb and @lemonig. I think what you're asking for is to allow interpolate to process areas beyond the boundaries of the input data. Like this yes? Same points except with larger bounding box.
> 
> ![Screenshot 2024-11-29 at 9 41 13 PM](https://private-user-images.githubusercontent.com/5403628/391057692-30c535fc-1eb9-4cd8-ac5a-37643f62eb7e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMxMDE3OTMsIm5iZiI6MTczMzEwMTQ5MywicGF0aCI6Ii81NDAzNjI4LzM5MTA1NzY5Mi0zMGM1MzVmYy0xZWI5LTRjZDgtYWM1YS0zNzY0M2Y2MmViN2UucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDJUMDEwNDUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjRkYTllY2Y3NzhlNGJjNDBkY2E1MDc5YzU5YWFjZGI5OTVlOTdhZWRlZmFiZmQ0ZDUxYjgyZDA0ZmVlNjk1MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.9qBUFNZOAhtQGnp75H7jfIbARSUxqUfRcECyEUPh-Dc)


Yes, that’s exactly how you described it. Could you tell me how you figured it out? Can you help solve this problem?

**Comment by lemonig** [comment]
> HI @mumispb and @lemonig. I think what you're asking for is to allow interpolate to process areas beyond the boundaries of the input data. Like this yes? Same points except with larger bounding box.
> 
> ![Screenshot 2024-11-29 at 9 41 13 PM](https://private-user-images.githubusercontent.com/5403628/391057692-30c535fc-1eb9-4cd8-ac5a-37643f62eb7e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMyMTgwMjEsIm5iZiI6MTczMzIxNzcyMSwicGF0aCI6Ii81NDAzNjI4LzM5MTA1NzY5Mi0zMGM1MzVmYy0xZWI5LTRjZDgtYWM1YS0zNzY0M2Y2MmViN2UucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDNUMDkyMjAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTQ0ZWZmNTYwYmQ4OGRjMThjM2JmNzlmNWVkN2IwMDZhYzZmNDExZjFiOWU1Y2QwMDViY2E1MTBjNTBlYzY0YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.25CJgf8fAVMctwDnreSiB_mFuu4eKF-rgGHMH86mAz8)

I already know how to generate the result of the right picture, thank you, @mumispb You can do that
![image](https://github.com/user-attachments/assets/f184b4b8-a1c6-4a18-a0be-382edd2a1947)
Interpolate in this area, Interpolate by yourself. That's it

**Comment by smallsaucepan** [comment]
Hi @lemonig.

> Could you tell me how you figured it out? Can you help solve this problem?

To make the right hand image I had to modify the Turf code to make the bounding box bigger. Will change this issue to be about adding an optional bbox option to give all users that level of control.

In the meantime, I thought of another workaround. All you would need to do is add four "dummy" points to your data, one at each corner of the bounding box you want to interpolate data within. That will trick the interpolate function into filling in the entire area.

Those four points should be as neutral as possible to not distort real points too much. You might need to experiment to find what's best. Perhaps try the average of all your points, or the minimum, for example.

It wouldn't be perfect, though should give reasonable results without having to wait for an update to Turf.

**Comment by mumispb** [comment]
I have used the exact same solution as @smallsaucepan , but I have to say that the values for these extra points can be a headache, because different approaches work for different cases. It can be useful to let the users of your app define a value for these points and let them experiment. I mean, if that's your use case. I have tried using 0, 1, minimum, average, average of nearest points.

**Comment by smallsaucepan** [comment]
@mumispb and @lemonig would either of you like to contribute a PR? Adding an optional `bbox` param to interpolate, and if provided using that rather than the bounding box of the data points?

Happy to do it if you don't have the time. Thought I'd offer the opportunity first though.

**Comment by lemonig** [comment]
> @mumispb and @lemonig would either of you like to contribute a PR? Adding an optional `bbox` param to interpolate, and if provided using that rather than the bounding box of the data points?
> 
> Happy to do it if you don't have the time. Thought I'd offer the opportunity first though.

I am very happy to do it, I think I can give it a try。and I suggest that IDW can add a search point `radius`, which can control the search radius of interpolation points when there are too many points。

**Comment by smallsaucepan** [comment]
Excellent. Thanks @lemonig.

> I suggest that IDW can add a search point `radius`, which can control the search radius of interpolation points when there are too many points

Do you mean this would be a new optional parameter? Would it keep the current behaviour (be backwards compatible) if called with default arguments?

**Comment by smallsaucepan** [comment]
Note resolving this will also resolve #1031.

**Comment by lemonig** [comment]
I have added PR to solve the problem of interpolating the specified area with `bbox` params.Please let me know if anything needs to be changed.

**Comment by smallsaucepan** [comment]
This was resolved by the merge of #2768
